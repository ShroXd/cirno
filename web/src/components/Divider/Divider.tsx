export const Divider = ({
  variant = 'horizontal',
  className = '',
}: {
  variant?: 'horizontal' | 'vertical'
  className?: string
}) => {
  if (variant === 'horizontal') {
    return <div className={`h-px bg-gray-300 ${className}`} />
  } else {
    return <div className={`h-full w-px bg-gray-300 ${className}`} />
  }
}
