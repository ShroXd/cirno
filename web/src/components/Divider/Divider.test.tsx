import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { describe, expect, it } from 'vitest'

import { Divider } from './Divider'

describe('Divider', () => {
  it('should render without a11y violations', async () => {
    const { container } = render(<Divider />)
    expect(await axe(container)).toHaveNoViolations()
  })

  it('should render correctly with horizontal variant', () => {
    const { container } = render(<Divider variant='horizontal' />)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render correctly with vertical variant', () => {
    const { container } = render(<Divider variant='vertical' />)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })
})
