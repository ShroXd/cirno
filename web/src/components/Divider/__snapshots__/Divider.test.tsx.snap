// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > should render correctly with horizontal variant 1`] = `
<div
  class="h-px bg-gray-300 "
/>
`;

exports[`Divider > should render correctly with vertical variant 1`] = `
<div
  class="h-full w-px bg-gray-300 "
/>
`;
